retrieve-logs/fetch_missing_matomo_requests.py

```diff
@@ -1,48 +1,99 @@
-from datetime import datetime, timedelta, timezone
+from datetime import date, datetime, time, timedelta, timezone
 import os
 import boto3
+import logging
 import time
-from _decimal import Decimal
 
+LOG_LEVEL = 'LOG_LEVEL'
 NUM_OF_DAYS = 'NUM_OF_DAYS'
+OUTPUT_FILENAME = 'OUTPUT_FILENAME'
+PERIOD_WIDTH_IN_SECONDS = 'PERIOD_WIDTH_IN_SECONDS'
 START_DATE = 'START_DATE'
 
+DATE_FORMAT = '%Y-%m-%d'
+FILENAME_SUFFIX = '_matomo_requests.json'
+MAX_REQUESTS = 10_000
 
-def print_unset_env_variable_error_and_exit(environment_variable):
-    print(environment_variable, " environment variable is not set.")
+_logger = None
+
+
+def get_logger():
+    global _logger
+    if not _logger:
+        logging.basicConfig(level=logging.INFO)
+        _logger = logging.getLogger(__name__)
+        _logger.setLevel(os.getenv(LOG_LEVEL, logging.INFO))
+    return _logger
+
+
+def log_too_many_requests_and_exit(period_start, period_end):
+    get_logger().error(
+            f'10000 requests received from the period {period_start} to {period_end}.'
```

Comment:
Can we interpolate `MAX_REQUESTS` into the error message instead of hard coding it?

Reply:
Good point. I'll fix this in #3 though I think.

Reply:
This should be fixed now. Though I've spotted another one... 😞